src/psyclone/parse/algorithm.py

```diff
@@ -201,14 +236,53 @@ def parse(self, alg_filename):
             # Nothing relevant found.
             raise ParseError(
                 "algorithm.py:parser:parse: Program, module, function or "
-                "subroutine not found in parse tree for file "
-                "'{0}'".format(alg_filename))
+                "subroutine not found in fparser2 parse tree.")
 
         self._unique_invoke_labels = []
         self._arg_name_to_module_name = OrderedDict()
+        self._arg_type_defns = dict()
```

Comment:
Please add a comment saying what this will hold.

Reply:
Added.